tests/TestKdbx4.cpp

```diff
@@ -403,3 +405,14 @@ void TestKdbx4::testCustomData()
     QCOMPARE(newEntry->customData()->value(customDataKey1), customData1);
     QCOMPARE(newEntry->customData()->value(customDataKey2), customData2);
 }
+
+QSharedPointer<Kdf> TestKdbx4::fastKdf(QSharedPointer<Kdf> kdf)
+{
+    kdf->setRounds(1);
+    kdf->processParameters({
+        {KeePass2::KDFPARAM_ARGON2_MEMORY, 1024},
+        {KeePass2::KDFPARAM_ARGON2_PARALLELISM, 1}
```

Comment:
You cannot just use Argon2 unconditionally. The test must not make assumptions about which part of changing the KDF triggers the KDBX 4 upgrade. Imagine we introduce a bug where only Argon2 triggers the upgrade, but not AES-KDF. With these changes we would never notice.

Reply:
This does not actually use Argon2, but iff Argon2 is used, then the parameters are applied. `AesKdf` ignores these, but the cipher is not changed.

Reply:
Actually, thinking more about it... You are just setting the parameters and not changing the actual KDF. But I would still prefer you add a check here.